File: PlaybackLab/TVOSSampleApp/TVOSSampleApp/Players/FullscreenPlayerViewController.m
//
//  FullscreenPlayerViewController.m
//  TVOSSampleApp
//
//  Copyright © 2016 Ooyala. All rights reserved.
//

#import "FullscreenPlayerViewController.h"
#import <OoyalaSDK/OOOoyalaPlayer.h>
#import <OoyalaSDK/OOPlayerDomain.h>
#import "PlayerSelectionOption.h"

@interface FullscreenPlayerViewController ()

@property (nonatomic) NSString *pcode;
@property (nonatomic) NSString *playerDomain;

@end

@implementation FullscreenPlayerViewController

- (void)viewDidLoad {
  [super viewDidLoad];

  self.pcode = self.option.pcode;
  self.playerDomain = self.option.domain;

  self.player = [[OOOoyalaPlayer alloc] initWithPcode:self.pcode
                                               domain:[[OOPlayerDomain alloc] initWithString:self.playerDomain]];
  
  [NSNotificationCenter.defaultCenter addObserver:self
                                         selector:@selector(notificationHandler:)
                                             name:nil
                                           object:self.player];
  
  [self.player setEmbedCode:self.option.embedCode];
  [self.player play];
}

- (void)notificationHandler:(NSNotification *)notification {
  if ([notification.name isEqualToString:OOOoyalaPlayerTimeChangedNotification]) {
    return;
  }
  
  NSLog(@"Notification Received: %@. state: %@. playhead: %f",
        notification.name,
        [OOOoyalaPlayerStateConverter playerStateToString:self.player.state],
        self.player.playheadTime);
}

@end

File: PlaybackLab/TVOSSwiftSampleApp/TVOSSwiftSampleApp/Players/FullscreenPlayerViewController.swift
//
//  FullscreenPlayerViewController.swift
//  TVOSSwiftSampleApp
//
//  Copyright © 2018 Ooyala. All rights reserved.
//

import Foundation

class FullscreenPlayerViewController: OOOoyalaTVPlayerViewController {
  
  var option : PlayerSelectionOption!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    player = OOOoyalaPlayer(pcode: option.pcode, domain: OOPlayerDomain(string: option.domain))
    NotificationCenter.default.addObserver(self, selector: #selector(notificationHandler(_:)), name: nil, object: player)
    player.setEmbedCode(option.embedCode)
    player.play()
  }
  
  @objc func notificationHandler(_ notification: Notification) {
    let name = notification.name
    if name == NSNotification.Name.OOOoyalaPlayerTimeChanged {
      return
    }
    
    NSLog("Notification Received: %@. state: %@. playhead: %f", name.rawValue,
          OOOoyalaPlayerStateConverter.playerState(toString: player.state()),
          player.playheadTime())
  }
  
}
